kotlinx-coroutines-core/common/src/CancellableContinuationImpl.kt

```diff
@@ -625,19 +621,40 @@ private object Active : NotCompleted {
 }
 
 /**
- * Base class for all [CancellableContinuation.invokeOnCancellation] handlers to avoid an extra instance
- * on JVM, yet support JS where you cannot extend from a functional type.
+ * Essentially the same as just a function from `Throwable?` to `Unit`.
+ * The only thing implementors can do is call [invoke].
+ * The reason this abstraction exists is to allow providing a readable [toString] in the list of completion handlers
+ * as seen from the debugger.
+ * Use [UserSupplied] to create an instance from a lambda.
+ * We can't avoid defining a separate type, because on JS, you can't inherit from a function type.
+ *
+ * @see InternalCompletionHandler for a very similar interface, but used for handling completion and not cancellation.
  */
-internal abstract class CancelHandler : CancelHandlerBase(), NotCompleted
-
-// Wrapper for lambdas, for the performance sake CancelHandler can be subclassed directly
-private class InvokeOnCancel( // Clashes with InvokeOnCancellation
-    private val handler: CompletionHandler
-) : CancelHandler() {
-    override fun invoke(cause: Throwable?) {
-        handler.invoke(cause)
+internal interface CancelHandler: NotCompleted {
+    /**
+     * Signal cancellation.
```

Comment:
Probably it's "Signal**s**" (as per our general rule that is not explicitly stated anywhere)